Tools/clinic/clinic.py

```diff
@@ -2495,11 +2552,22 @@ def __init__(
         global clinic
         clinic = self
 
-    def add_include(self, name: str, reason: str) -> None:
-        if name in self.includes:
-            # Mention a single reason is enough, no need to list all of them
-            return
-        self.includes[name] = reason
+    def add_include(self, name: str, reason: str, condition: str | None = None) -> None:
```

Comment:
I would probably make condition a keyword-only parameter on this and the other `add_include()` method if I were writing it (but I won't insist :)